src/main/java/ch/uzh/ifi/hase/soprafs22/game/Game.java

```diff
@@ -79,7 +82,20 @@ public GameMap getGameMap() {
     }
 
     public int nextTurn() {
-        return ++turnNumber;
+        // TODO test for real once turns can finish through the web.
+        ++turnNumber;
+        for (IPlayer player : playerMap.values()) {
+            String token = player.getToken();
+            if (isPlayersTurn(token))
+                socketMessage.convertAndSend("/topic/game/token" + token, "yes_turn");
```

Comment:
we shouldn't send a message from the Game class via socketmessage to our client.
The Game should return something to the GameService.
The GameService returns something to the GameController.
The GameController uses socketmessage to speak to the client

Reply:
If we use the return value there is no guarantee that classes receiving it will wire it appropriately to the websocket.
I've done it as you asked, but please make sure that **every** time there is a next turn, it makes it to the controller (e.g. attack ends turn, move ends turn, etc.)